File: backend/src/main/java/org/ftclub/cabinet/dto/CabinetInfoPagenationDto.java
package org.ftclub.cabinet.dto;

import lombok.*

@AllArgsConstructor
@NoArgsConstructor
@Data
public class CabinetInfoPagenationDto {
    /**
     * 기존 cabinetInfoPagenationDto를 변형했습니다.
     */
    private CabinetInfoResponseDto[] result;
    private Long totalLength;
}

File: backend/src/admin/dto/cabinet.info.pagenation.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { CabinetInfoDto } from './cabinet.info.dto';

export class CabinetInfoPagenationDto {
  @ApiProperty({
    description: '캐비넷 정보 배열',
    type: [CabinetInfoDto],
  })
  result: CabinetInfoDto[]; // 캐비넷 정보 배열

  @ApiProperty({
    description: 'DB에 저장된 총 결과의 길이',
    example: 42,
  })
  total_length: number; // DB에 저장된 총 결과의 길이
}
